sdk/src/client/api/block_builder/input_selection/core/mod.rs

```diff
@@ -403,10 +406,134 @@ impl InputSelection {
             return Err(Error::InvalidOutputCount(self.outputs.len()));
         }
 
+        self.validate_transitions()?;
+
         Ok(Selected {
             inputs: Self::sort_input_signing_data(self.selected_inputs, &self.outputs, Some(self.timestamp))?,
             outputs: self.outputs,
             remainder,
         })
     }
+
+    fn validate_transitions(&self) -> Result<(), Error> {
+        let mut input_native_tokens_builder = NativeTokensBuilder::new();
+        let mut output_native_tokens_builder = NativeTokensBuilder::new();
+        let mut input_aliases = Vec::new();
+        let mut input_chains_foundries = hashbrown::HashMap::new();
+        let mut input_foundries = Vec::new();
+        let mut input_nfts = Vec::new();
```

Comment:
consider using with capacity so it doesn't have to reallocate?

Reply:
We don't know the number in advance

Reply:
Upper limit is 128, but can often also stay completely empty